packages/coreutils/src/poll.ts

```diff
@@ -380,6 +378,7 @@ export class Poll<T = any, U = any> implements IDisposable, IPoll<T, U> {
 
     // Update poll state.
     const last = this.state;
+    const pending = this._tick;
```

Comment:
Ah, from conversation in gitter, I'm understanding now that the await above may make pending stale, but here we need pending to be the *current* `this._tick`. Is that the fix?

Reply:
Yes, exactly that.